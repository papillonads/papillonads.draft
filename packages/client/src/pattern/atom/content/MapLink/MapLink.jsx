import React from 'react'
import cx from 'classnames'
import { mapLink, mapLinkHeading, mapLinkImage } from './MapLink.scss'
import { mapLinkDefaultProps, mapLinkPropTypes } from './MapLink.props'

export function MapLink({ className }) {
  return (
    <div className={cx(className, mapLink)}>
      <h3 className={mapLinkHeading}>
        <a href="kijkinuwwijk.html">Kijk in je Wijk</a>
      </h3>
      <a href="kijkinuwwijk.html" className={mapLinkImage}>
        <img alt="" src="https://s.marktplaats.com/z/res/images/map_link.png?break" />
      </a>
    </div>
  )
}

MapLink.defaultProps = mapLinkDefaultProps

MapLink.propTypes = mapLinkPropTypes
