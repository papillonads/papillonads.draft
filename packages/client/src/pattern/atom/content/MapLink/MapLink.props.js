import { string } from 'prop-types'

export const mapLinkDefaultProps = {
  className: '',
}

export const mapLinkPropTypes = {
  className: string,
}
