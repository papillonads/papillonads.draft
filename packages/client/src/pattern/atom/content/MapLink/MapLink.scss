@import '../../../../sass/mixin';

.mapLink {
  &Heading {
    &Anchor {
      color: #2d3c4d;

      @include leftColumnTitle;
    }
  }

  &Image {
    border-radius: 33px;
    box-shadow: 0 1px 1px 0 rgba(0, 0, 0, 0.14);
    cursor: pointer;
    width: 100%;
    max-width: 190px;
  }
}
