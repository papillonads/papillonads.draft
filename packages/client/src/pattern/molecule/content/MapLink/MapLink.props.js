import { string, shape } from 'prop-types'

export const mapLinkDefaultProps = {
  className: '',
}

export const mapLinkPropTypes = {
  className: string,
  data: shape({}).isRequired,
}
