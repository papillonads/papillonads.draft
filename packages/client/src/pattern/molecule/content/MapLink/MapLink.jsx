import React from 'react'
import cn from 'classnames'
import { mapLinkDefaultProps, mapLinkPropTypes } from './MapLink.props'

const MapLink = ({ className }) => {
  const mapLinkClassList = cn(className)

  return <div className={mapLinkClassList} />
}

MapLink.defaultProps = mapLinkDefaultProps

MapLink.propTypes = mapLinkPropTypes

export default MapLink
