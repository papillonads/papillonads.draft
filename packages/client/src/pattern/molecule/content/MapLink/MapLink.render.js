import React from 'react'
import MapLink from './MapLink'
import homePageData from '../../../../../../server/src/data/page/home'

const defaultRender = <MapLink data={homePageData.content.mapLink} />

export { defaultRender }
