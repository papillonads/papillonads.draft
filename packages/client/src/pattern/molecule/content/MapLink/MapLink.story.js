import { storiesOf } from '@storybook/react'
import { defaultRender } from './MapLink.render'
import { getInfo } from '../../../../../../../.storybook/library'

const defaultRenderInfo = getInfo('default')

storiesOf('Patttern/Molecule/Content/MapLink', module).add(defaultRenderInfo.text, () => defaultRender, defaultRenderInfo.parameters)
