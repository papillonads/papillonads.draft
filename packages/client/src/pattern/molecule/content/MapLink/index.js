import MapLink from './MapLink'

export default MapLink
