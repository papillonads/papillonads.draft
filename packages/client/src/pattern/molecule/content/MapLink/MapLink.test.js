import { defaultRender } from './MapLink.render'

describe('<MapLink />', () => {
  describe('Snaphot', () => {
    test('must match defaultRender', () => {
      expect(global.renderToJSON(defaultRender)).toMatchSnapshot()
    })
  })
})
