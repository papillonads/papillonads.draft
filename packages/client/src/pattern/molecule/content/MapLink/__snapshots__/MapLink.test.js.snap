// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<MapLink /> Snaphot must match defaultRender 1`] = `
<div
  className="mapLink"
/>
`;
